hibernate-core/src/main/java/org/hibernate/dialect/TeradataDialect.java

```diff
@@ -120,6 +122,24 @@ public TeradataDialect(int version) {
 
 	}
 
+	@Override
+	public SqlTypeDescriptor resolveSqlTypeDescriptor(
+			int jdbcTypeCode,
+			int precision,
+			int scale,
+			SqlTypeDescriptorRegistry sqlTypeDescriptorRegistry) {
+		switch ( jdbcTypeCode ) {
+			case Types.BIT:
+				return sqlTypeDescriptorRegistry.getDescriptor( Types.BOOLEAN );
```

Comment:
Can't we get rid of stuff like this now?

Reply:
I don't think we can because this is for supporting native queries.

Reply:
Oh it's for reflecting on the metadata of the native query JDBC result set and guessing the Hibernate types?

OK, that makes sense.

Reply:
Correct